src/hwProvider.ts

```diff
@@ -31,47 +38,117 @@ export class HWProvider {
      * Creates transport and initialises ledger app.
      */
     async init(): Promise<boolean> {
+        if (this.isInitialized()) {
+            return true;
+        }
+
         try {
-            const transport = await this.getTransport();
-            this._hwApp = new LedgerApp(transport);
+            this._transport = await this.getTransport();
+            this._hwApp = new LedgerApp(this._transport);
 
-            return true;
+            return this.isInitialized();
         } catch (error) {
             console.error("Provider initialization error", error);
             return false;
         }
     }
 
     async getTransport(): Promise<Transport> {
-        let webUSBSupported = await TransportWebUSB.isSupported();
-        webUSBSupported =
-            webUSBSupported &&
-            platform.name !== "Opera";
+        if (this._transport) {
+            return this._transport;
+        }
+
+        const isLedgerSupported = await this.isLedgerTransportSupported();
+
+        if (!isLedgerSupported) {
+            throw Error("Ledger is not supported");
+        }
+
+        try {
+            const webUSBSupported = await this.isWebUSBSupported();
+
+            if (webUSBSupported) {
+                console.log('Web USB Transport selected');
+                this._transportType = TransportTypeEnum.USB;
```

Comment:
Object state isn't usually mutated in `get`-like functions. An alternative would have been to return a tuple from the function (both the transport object and the type).